web/src/main/java/se/inera/intyg/intygstjanst/web/service/impl/TextSearchFilterServiceImpl.java

```diff
@@ -0,0 +1,127 @@
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.intygstjanst.web.service.impl;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.infra.sjukfall.dto.SjukfallEnhet;
+import se.inera.intyg.intygstjanst.web.service.CalculatePatientAgeService;
+import se.inera.intyg.intygstjanst.web.service.CalculatePatientGenderService;
+import se.inera.intyg.intygstjanst.web.service.DiagnosisChapterService;
+import se.inera.intyg.intygstjanst.web.service.TextSearchFilterService;
+
+@Service
+public class TextSearchFilterServiceImpl implements TextSearchFilterService {
+
+    private final CalculatePatientAgeService calculatePatientAgeService;
+    private final CalculatePatientGenderService calculatePatientGenderService;
+    private final DiagnosisChapterService diagnosisChapterService;
+
+    public TextSearchFilterServiceImpl(CalculatePatientAgeService calculatePatientAgeService,
+        CalculatePatientGenderService calculatePatientGenderService, DiagnosisChapterService diagnosisChapterService) {
+        this.calculatePatientAgeService = calculatePatientAgeService;
+        this.calculatePatientGenderService = calculatePatientGenderService;
+        this.diagnosisChapterService = diagnosisChapterService;
+    }
+
+    @Override
+    public List<SjukfallEnhet> filterList(List<SjukfallEnhet> sickLeaves, String textSearch) {
+        if (textSearch == null || textSearch.isEmpty()) {
+            return sickLeaves;
+        }
+        return filterOnTextSearch(sickLeaves, textSearch);
+    }
+
+    @Override
+    public boolean filter(SjukfallEnhet sickLeave, String textSearch) {
+        if (textSearch == null || textSearch.isEmpty()) {
+            return true;
+        }
+        return textSearchMatchesAnyField(sickLeave, textSearch);
+    }
+
+    private List<SjukfallEnhet> filterOnTextSearch(List<SjukfallEnhet> sickLeaves, String textSearch) {
+        return sickLeaves.stream()
+            .filter(sickLeave -> textSearchMatchesAnyField(sickLeave, textSearch))
+            .collect(Collectors.toList());
+    }
+
+    private boolean textSearchMatchesAnyField(SjukfallEnhet sickLeave, String textSearch) {
+        return searchMatchesAge(sickLeave, textSearch) || searchMatchesPatientName(sickLeave, textSearch) || searchMatchesPatientGender(
```

Comment:
Very subjective comment.. but if you put each "search.." on a new line it will be much easier to read the code.